lte/gateway/python/magma/mobilityd/ip_address_man.py

```diff
@@ -432,6 +436,19 @@ def _try_set_recycle_timer(self):
                 else:
                     self._recycle_reaped_ips()
 
+    def _allocate_static_ip(self, sid: str) -> IPDesc:
+        """
+        Check if static IP allocation is enabled and then check
+        subscriber DB for assigned static IP for the SID
+        """
+        if self.static_ip_enabled is False:
+            return None
+        ip_addr = subscriberdb_client.get_subscriber_ip(sid)
+        if ip_addr is None:
+            return None
+        return IPDesc(ip=ip_addr, state=IPState.ALLOCATED,
+                      sid=sid, ip_block=ip_addr, ip_type=IPType.STATIC)
+
```

Comment:
this logic belongs to the IPAllocator, not this class. 2 different options here:

1. If static IP enabled means static IP *only*, then create a new implementation of IPAllocator for static IP configuration.
2. If static IP enabled means allocate a static IP when it's configured and another method if it's not, then create a new IPAllocator implementation which can wrap a "fallback" IP Allocator.

High level point here is that it's not this component's responsibility to know about the details of IP allocation, static or otherwise

Reply:
done, Added wrapper class.